assets/js/stripe-payment-request.js

```diff
@@ -453,6 +453,53 @@ jQuery( function( $ ) {
 			} );
 		},
 
+		getSelectedVariationCachedData: function () {
```

Comment:
I think we should minimize introducing code outside the scope of Payment Request buttons in this file and in the extension in general.

Note that when changing the dropdowns the product changes according to the selected attributes. So there must be code that can get the right variation.

<img src="https://user-images.githubusercontent.com/1025173/141597485-98355633-6789-4845-a072-1bbedc90d403.gif" width="320" />

And that code is here: 
 https://github.com/woocommerce/woocommerce/blob/fd3dfdbf3e05ce87138831d2f8eff3734eeb4572/plugins/woocommerce/assets/js/frontend/add-to-cart-variation.js#L239

Which is called here if the variation is found https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/assets/js/frontend/add-to-cart-variation.js#L210 

I think we should leverage the existing events attached to the form and use it to get the selected variation and with that we could hide/unhide the PRB accordingly.

The code can be simplified to just:


```js
if ( $('.variations_form').length ) {
  $('.variations_form').on('found_variation.wc-variation-form', function (evt, variation) {
    if (variation.is_in_stock) {
      wc_stripe_payment_request.unhidePaymentRequestButton();
    } else {
      wc_stripe_payment_request.hidePaymentRequestButton();
    }
  });
}
```

Reply:
I didn't notice that there was existing code that already retrieved the right variation 🤦  

Good call. That definitely looks cleaner 👍